src/Network/NWContentContext.cs

```diff
@@ -0,0 +1,232 @@
+		}
+
+		delegate void ProtocolIterator (IntPtr block, IntPtr definition, IntPtr metadata);
+		static ProtocolIterator static_ProtocolIterator = TrampolineProtocolIterator;
+		
+		[MonoPInvokeCallback (typeof (ProtocolIterator))]
+		static unsafe void TrampolineProtocolIterator (IntPtr block, IntPtr definition, IntPtr metadata)
+		{
+			var descriptor = (BlockLiteral *) block;
+			var del = (Action<NWProtocolDefinition,NWProtocolMetadata>) (descriptor->Target);
+			if (del != null){
+				var pdef = definition == IntPtr.Zero ? null : new NWProtocolDefinition (definition, owns: true);
+				var meta = metadata == IntPtr.Zero ? null : new NWProtocolMetadata (metadata, owns: true);
+
+				del (pdef, meta);
+				
+				pdef?.Dispose ();
+				meta?.Dispose ();
+				if (pdef != null)
+					pdef.Dispose ();
+				if (meta != null)
+					meta.Dispose ();
+			}
+		}
+		
+		[TV (12,0), Mac (10,14), iOS (12,0)]
+		[DllImport (Constants.NetworkLibrary)]
+		static extern unsafe void nw_content_context_foreach_protocol_metadata (IntPtr handle, void *callback);
+		
+		[TV (12,0), Mac (10,14), iOS (12,0)]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public void IterateProtocolMetadata (Action<NWProtocolDefinition,NWProtocolMetadata> callback)
+		{
+			unsafe {
+			        BlockLiteral *block_ptr_handler;
+			        BlockLiteral block_handler;
+			        block_handler = new BlockLiteral ();
+			        block_ptr_handler = &block_handler;
+			        block_handler.SetupBlockUnsafe (static_ProtocolIterator, callback);
+			
+			        nw_content_context_foreach_protocol_metadata (GetHandle(), (void*) block_ptr_handler);
+			        block_ptr_handler->CleanupBlock ();
+			}
+		}
+
+		//
+		// Use this as a parameter to NWConnection.Send's with all the default properties
+		// ie: NW_CONNECTION_DEFAULT_MESSAGE_CONTEXT, use this for datagrams
+		static NWContentContext defaultMessage;
+		public static NWContentContext DefaultMessage {
+			get {
+				if (defaultMessage == null)
+					defaultMessage = MakeGlobal (Marshal.ReadIntPtr (Dlfcn.dlsym (Libraries.Network.Handle, "_nw_content_context_default_message")));
+				
+				return defaultMessage;
+			}
+		}
+
+		// Use this as a parameter to NWConnection.Send's to indicate that no more sends are expected
+		// (ie: NW_CONNECTION_FINAL_MESSAGE_CONTEXT)
+		static NWContentContext finalMessage;
+		public static NWContentContext FinalMessage {
+			get {
+				if (finalMessage == null)
+					finalMessage = MakeGlobal (Marshal.ReadIntPtr (Dlfcn.dlsym (Libraries.Network.Handle, "_nw_content_context_final_send")));
+				return finalMessage;
+			}
+		}
+
+		// This sending context represents the entire connection
+		// ie: NW_CONNECTION_DEFAULT_STREAM_CONTEXT
+		static NWContentContext defaultStream;
+		public static NWContentContext DefaultStream {
+			get {
+				if (defaultStream == null)
+					defaultStream = MakeGlobal (Marshal.ReadIntPtr (Dlfcn.dlsym (Libraries.Network.Handle, "_nw_content_context_default_stream")));
```

Comment:
Why do you need to make these objects global when they're stored in static variables anyway?

Reply:
To prevent someone calling `Dispose()` and ruining the global instance for everyone else.